### PR TITLE
chore(cli): rename "app deploy" to "svc deploy"

### PR DESCRIPTION
**Pull request by efekarakus**
This change has the following fixes:
* moves the `selector.go` file to its own package and is now used in "svc deploy".
* fixes the Makefile for the `describe` pkg mocks.
* updates `app deploy` to `svc deploy`. 

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: internal/pkg/cli/interfaces.go [hunk 1]
```diff
@@ -325,3 +325,18 @@ type askExecutor interface {
...
+	Environment(prompt, help, app string) (string, error)
+}
+
+type configSelector interface {
+	appEnvSelector
+	Service(prompt, help, app string) (string, error)
+}
+
+type wsSelector interface {
+	appEnvSelector
```
[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/interfaces.go:340]: oh I see, ok the name of my interface is confusing 😛. By `wsSelector` I meant  an "app, env, and service" selector but where the service is selected from the ws.

What do you think of naming the two interfaces like this instead: `appEnvWsSvcSelector` and `appEnvConfigSvcSelector`?

[the remark this answers is not in the document]
Comment (iamhopaul123) [reply internal/pkg/cli/interfaces.go:340]: I think i prefer the previous one. Maybe just leave it as what it is now.

File: internal/pkg/cli/svc_deploy.go [hunk 2]
```diff
@@ -36,66 +37,68 @@ var (
...
-	appCFSvc         cloudformation.CloudFormation
-	sessProvider     sessionProvider
+	store        store
+	ws           wsAppReader
+	ecr          ecrService
+	docker       dockerService
+	s3           artifactUploader
+	cmd          runner
+	addons       templater
+	appCFN       projectResourcesGetter
```
[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy.go]: Good catch! I renamed a few of the other ones too like `wsSvcManifestReader`

File: internal/pkg/cli/svc_deploy.go [hunk 3]
```diff
@@ -503,20 +470,20 @@ func (o *appDeployOpts) showAppURI() error {
 
 // BuildAppDeployCmd builds the `app deploy` subcommand.
 func BuildAppDeployCmd() *cobra.Command {
```
[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy.go]: Done!

File: internal/pkg/cli/svc_deploy_test.go [hunk 4]
```diff
@@ -115,124 +115,41 @@ func TestAppDeployOpts_Validate(t *testing.T) {
 
 func TestAppDeployOpts_Ask(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy_test.go]: Done!

File: internal/pkg/cli/svc_deploy_test.go [hunk 5]
```diff
@@ -297,7 +208,7 @@ image:
 		wantPath string
 		wantErr  error
 	}{
-		"should return error if workspaceService ReadFile returns error": {
+		"should return error if ws ReadFile returns error": {
 			inputApp: "appA",
```
[the remark this answers is not in the document]
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy_test.go]: Done! fixed the titles in the tests

**Pull request merged**